torchvision/datasets/mnist.py

```diff
@@ -265,6 +265,9 @@ def __init__(self, root: str, split: str, **kwargs: Any) -> None:
         super(EMNIST, self).__init__(root, **kwargs)
         self.classes = self.classes_split_dict[self.split]
 
+        if self.target_transform is None and self.split == 'letters':
+            self.target_transform = lambda x: x - 1
```

Comment:
To me I think it might be preferable to change `classes_split_dict` case for `letters`, if it's not compatible with the current annotations.

I think that implicitly changing in the `target_transform` the labels might be misleading to users, and would make previously trained models to not be valid anymore.

Reply:
I did think of that initially while fixing this, but the issue is the class to letter mapping is still offset by 1. The only other solution would be to add a dummy letter to position 0 to fix the class offset. For ex: 1 is A while loading the dataset, but classes_split_dict would return b. I added this fix since transforms is a cleaner approach, but adding a dummy letter to the list is preferred, I'm happy to change it to that too.

Reply:
> add a dummy letter to position 0

I think this would be the preferred approach IMO. Maybe add something like `__unused__` ?

Reply:
I picked up @vballoli's great work to make the final changes and merge.

@fmassa I used `N/A` instead because COCO uses this value for the unused categories.

Reply:
Using `N/A` is fine with me